File: backend/src/main/java/com/gvntmc/backend/entity/Article.java
package com.gvntmc.backend.entity;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Getter
@Setter
@Document(collection = "articles")
public class Article {

    @Id
    private String id;

    private String title;
    private String titleEn;

    @Indexed(unique = true)
    private String slug;

    private String excerpt;
    private String excerptEn;
    private String content;
    private String contentEn;
    private String category;
    private String categoryEn;
    private String emoji;
    private String bg;
    private int    readTime;
    private boolean published = false;

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;
}

File: lib/models/Article.ts
import { Schema, model, models } from "mongoose";

export interface IArticle {
  _id?: string;
  title: string;
  titleEn: string;
  slug: string;
  excerpt: string;
  excerptEn: string;
  content: string;
  contentEn: string;
  category: string;
  categoryEn: string;
  emoji: string;
  bg: string;
  readTime: number;
  published: boolean;
  createdAt?: Date;
  updatedAt?: Date;
}

const ArticleSchema = new Schema<IArticle>(
  {
    title:       { type: String, required: true },
    titleEn:     { type: String, default: "" },
    slug:        { type: String, required: true, unique: true },
    excerpt:     { type: String, required: true },
    excerptEn:   { type: String, default: "" },
    content:     { type: String, default: "" },
    contentEn:   { type: String, default: "" },
    category:    { type: String, default: "Tin tức" },
    categoryEn:  { type: String, default: "News" },
    emoji:       { type: String, default: "📰" },
    bg:          { type: String, default: "linear-gradient(135deg,#e8f0fb,#d0e4f8)" },
    readTime:    { type: Number, default: 3 },
    published:   { type: Boolean, default: false },
  },
  { timestamps: true }
);

ArticleSchema.index({ slug: 1 }, { unique: true });
ArticleSchema.index({ category: 1 });
ArticleSchema.index({ published: 1 });
ArticleSchema.index({ createdAt: -1 });

export const ArticleModel = models.Article || model<IArticle>("Article", ArticleSchema);
